src/components/Tables/Table.tsx

```diff
@@ -13,22 +13,21 @@ import {
 } from "@fortawesome/free-solid-svg-icons";
 import IconButton from "@mui/material/IconButton/IconButton";
 import Icon from "@/components/Icons/Icon";
+import { Filter, headerLabelFromKey, textFilter } from "@/components/Tables/Filters";
 
-export interface Datum {
-    [headerKey: string]: string | number | boolean | null;
-}
-
-export type TableHeaders = [string, string][];
+export type TableHeaders<Data> = readonly (readonly [keyof Data, string])[];
 
-export interface Row {
+export interface Row<Data> {
     rowId: number;
-    data: Datum;
+    data: Data;
 }
 
-export type ColumnDisplayFunction = (row: Row) => ReactNode;
-export type TableColumnDisplayFunctions = { [headerKey: string]: ColumnDisplayFunction };
+export type ColumnDisplayFunction<T> = (data: T) => ReactNode;
```

Comment:
I much prefer ReactElement (and then scattering fragments around) rather than ReactNode, where possible, but might be a reason for this I'm not aware of!